### DISCUSSION TITLE
Theia IDE 1.52 preview testing

### DISCUSSION THREAD

**Discussion by sgraband** [eclipse-theia/theia#13967] [Release Announcements] [answered]
The new version 1.52.0 of the Theia IDE is available on the preview channel now, please join the preview testing! You can download it here: [Linux](https://download.eclipse.org/theia/ide-preview/1.52.0/linux/TheiaIDE.AppImage), [Mac](https://download.eclipse.org/theia/ide-preview/1.52.0/macos/TheiaIDE.dmg), [Windows](https://download.eclipse.org/theia/ide-preview/1.52.0/windows/TheiaIDESetup.exe).. You can update your existing installation by setting the preference updates.channel to preview. Please respond here when you were able to test the preview without finding blockers, by commenting with a ✔️. If you find any issues, please mention them in this thread and report them as an issue once confirmed by other testers.

**Comment by tsmaeder** [comment]
I haven't really noticed any noticeable new bugs, but I get a bunch of errors in the browser logs:

```
Uncaught Error: Cannot read properties of undefined (reading 'keyCode')

TypeError: Cannot read properties of undefined (reading 'keyCode')
    at d.toKeybinding (bundle.js:2:7344410)
    at bundle.js:2:7343684
    at Array.map (<anonymous>)
    at d.getDispatchChords (bundle.js:2:7343627)
    at Le.softDispatch (bundle.js:2:6926144)
    at n._onKeyDown (bundle.js:2:6046908)
    at m.value (bundle.js:2:6043521)
    at Q._deliver (bundle.js:2:3966399)
    at Q._deliverQueue (bundle.js:2:3966505)
    at Q.fire (bundle.js:2:3966935)
    at bundle.js:2:3954382
(anonymous) @ bundle.js:2
```

```
bundle.js:2 Uncaught (in promise) Error: INVALID tab
    at u.acceptTabOperation (/C:/Users/thomas/AppData/Local/Programs/TheiaIDE/resources/app/lib/frontend/C:\Users\thomas\AppData\Local\Programs\TheiaIDE\resources\app\lib\backend\339.js:1:519975)
    at t.TabsExtImpl.$acceptTabOperation (/C:/Users/thomas/AppData/Local/Programs/TheiaIDE/resources/app/lib/frontend/C:\Users\thomas\AppData\Local\Programs\TheiaIDE\resources\app\lib\backend\339.js:1:522339)
    at /C:/Users/thomas/AppData/Local/Programs/TheiaIDE/resources/app/lib/frontend/C:\Users\thomas\AppData\Local\Programs\TheiaIDE\resources\app\lib\backend\366.js:1:35962
    at process.processTicksAndRejections (VM1720 task_queues:95:5)
    at async c.handleRequest (/C:/Users/thomas/AppData/Local/Programs/TheiaIDE/resources/app/lib/frontend/C:\Users\thomas\AppData\Local\Programs\TheiaIDE\resources\app\lib\backend\638.js:1:46979)
```

and 

```
bundle.js:2 2024-07-31T13:12:03.854Z root ERROR onWillSave listeners should provide edits, not directly alter the document.
```

**Reply by tsmaeder** [reply]
@msujew didn't we fix some of the tab problems for 1.52.0?

**Reply by msujew** [reply]
The change https://github.com/eclipse-theia/theia/pull/13812 was already included in 1.51.

**Reply by sgraband** [reply]
I can also reproduce the issue described above.

**Comment by rschnekenbu** [comment]
I don't see any other issues currently.

**Reply by sgraband** [reply]
Same for me :+1:

**Reply by JonasHelming** [reply]
same for me

**Reply by JonasHelming** [reply]
@tsmaeder It starts MUCH faster for me, as you anticipated

**Comment by tsmaeder** [comment]
I opened issues #13987, #13988 and #13990. It has happened to me that the typescript plugin did not update the errors in an open editor anymore and I had to reopen it. This could well be related to #13987. However, I don't think this is worse than in previous releases.
My proposal would be to go ahead with promoting the IDE release and to have another look at the the tabs issue for the next release. If we find a magic fix fast, we can always do a patch release. 
Thoughts? Volunteers?

**Reply by JonasHelming** [reply]
I think one of the links is wrong (13989)

**Reply by tsmaeder** [reply]
Fixed the links.

**Reply by sgraband** [reply]
I would promote to the stable channel later today, if there are no objections.

**Reply by tsmaeder** [reply]
Sounds good to me.

**Reply by sgraband** [reply, marked as answer]
1.52.0 is promoted to the stable channel

**Comment by JonasHelming** [comment]
[News & Noteworthy](https://eclipsesource.com/blogs/2024/08/06/eclipse-theia-1-52-release-news-and-noteworthy/)
